app/controllers/api/v1/users_controller.rb

```diff
@@ -3,13 +3,29 @@ module V1
     class UsersController < Api::V1::ApiController
       helper_method :user
 
+      before_action :password_check, only: :update
+
       def show; end
 
+      def update
+        user.update!(user_params)
+      end
+
       private
 
       def user
         @user ||= current_user
       end
+
+      def password_check
+        return if !user_params[:password] || user.valid_password?(params[:password_check])
```

Comment:
You are requiring the old password to update the password, right? If that is the case maybe you can extract the part after the if to make it more readable. What do you think?

Reply:
I don't think I understood

Reply:
Maybe you can do something of the sorts of:
```suggestion
        return if updating_password? && has_current_password?
```
Or rename `password_check` to something like `check_for_password_update`.

Reply:
password_check  is a param that comes in the body and should equal the user password, I don't think I fully understand your suggestion